### PR TITLE
fix: update documentation for support matrix

### PR DESCRIPTION
**Pull request by matttrach**
Update support matrix information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,27 +97,33 @@ See [test process](docs/test-process.md) for details on release testing (_Terraf
 Branching the Provider
 ---------------------------
 
-The provider is branched into three release lines with major version alignment with Rancher 2.6, 2.7, and 2.8. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6, the `release/v3` branch with 3.0.0+ is aligned with Rancher 2.7, and the `master` branch with 4.0.0+ is aligned with Rancher 2.8. The lifecycle of each major provider version is aligned with the lifecycle of each Rancher minor version. For example, provider versions 4.0.x which are aligned with Rancher 2.8.x will only be actively maintained until the EOM for Rancher 2.8.x and supported until EOL for Rancher 2.8.x.
+This provider is branched in correlation with minor versions of Rancher: 2.8, 2.9, etc.
```
Comment (jiaqiluo) [inline README.md:100]: Not sure what it looks like in your local developing environment, [Here](https://github.com/matttrach/terraform-provider-rancher2/blob/update-matrix/README.md#branching-the-provider) is what this file looks like in GitHub.  GitHub renders those lines in one paragraph and shows one space for multiple spaces. 

You may need to add empty lines to make it display as a multi-line paragraph in GH. The same rule applies to another file in the PR.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -97,27 +97,33 @@ See [test process](docs/test-process.md) for details on release testing (_Terraf
...
+  you must have a separate configuration in a separate dir for each provider.
+Otherwise, Terraform will overwrite the `.tfstate` file every time you switch versions.
 
 Releasing the Provider
 ---------------------------
 
 As of Terraform 2.0.0 and 3.0.0, the provider is tied to Rancher minor releases but can be released 'out of band' within that minor version.
 For example, 4.0.0 will be released 1-2 weeks after Rancher 2.8.x and fixes and features in the 4.0.0 release will be supported for clusters provisioned via Terraform on Rancher 2.8.x.
 A critical bug fix can be released 'out of band' as 4.0.1 and backported to `release/v3` as 3.0.1.
-A new feature can also be released 'out of band' as 4.1.0 but not backported.
```
Comment (jiaqiluo) [inline README.md:120]: Yeah, I agree to drop this one. Out-of-band release is cut only for bug fixing when it is necessary.

File: README.md (hunk 1 above)
Comment (matttrach) [reply README.md:100]: yeah, the change here was not to change it in GH, but to be able to read it in pure markdown without having to wrap the lines. For me it is just a readability issue, but it also makes the diffs cleaner and smaller in the future.

File: README.md [hunk 3]
```diff
@@ -97,27 +97,33 @@ See [test process](docs/test-process.md) for details on release testing (_Terraf
 Branching the Provider
 ---------------------------
 
-The provider is branched into three release lines with major version alignment with Rancher 2.6, 2.7, and 2.8. The `release/v2` branch with 2.0.0+ is aligned with Rancher 2.6, the `release/v3` branch with 3.0.0+ is aligned with Rancher 2.7, and the `master` branch with 4.0.0+ is aligned with Rancher 2.8. The lifecycle of each major provider version is aligned with the lifecycle of each Rancher minor version. For example, provider versions 4.0.x which are aligned with Rancher 2.8.x will only be actively maintained until the EOM for Rancher 2.8.x and supported until EOL for Rancher 2.8.x.
+This provider is branched in correlation with minor versions of Rancher: 2.8, 2.9, etc.
+The `release/v3` branch with 3.0.0+ is aligned with Rancher 2.7,
+  the `release/v4` branch with 4.0.0+ is aligned with Rancher 2.8,
+  and the `master` branch with 5.0.0+ is aligned with Rancher 2.9.
```
Comment (jiaqiluo) [inline README.md:103]: nit: maybe removing the leading spaces?

File: README.md [hunk 4]
```diff
@@ -97,27 +97,33 @@ See [test process](docs/test-process.md) for details on release testing (_Terraf
...
 * We can follow semver
-* We can cut patch/minor versions on an as-needed basis to fix bugs or add new resources 
+* We can cut patch/minor versions on an as-needed basis to fix bugs or add new resources
 * We have 'out of band' flexibility and are only tied to releasing a new version of the provider when we get a new 2.x Rancher minor version.
 
 See the [compatibility matrix](docs/compatibility-matrix.md) for details.
 
-If you are using Terraform to provision clusters on instances of Rancher 2.7 and 2.8, you must have a separate configuration in a separate dir for each provider. Otherwise, Terraform will overwrite the `.tfstate` file every time you switch versions.
+If you are using Terraform to provision clusters on instances of Rancher 2.7 and 2.8,
+  you must have a separate configuration in a separate dir for each provider.
```
Comment (jiaqiluo) [inline README.md:118]: nit: maybe removing the leading spaces?

**Review by jiaqiluo — APPROVED** [review]
LGMT.  
I left a few comments but they should not block the merging of the PR.

**Pull request merged**